pkg/util/nfacct/nfacct.go

```diff
@@ -0,0 +1,274 @@
+}
+
+// Get is part of the interface.
+func (r *Runner) Get(name string) (*Counter, error) {
+	req := r.newRequest(cmdGet)
+	req.AddData(nl.NewRtAttr(attrName, nl.ZeroTerminated(name)))
+	msgs, err := r.execute(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var counter *Counter
+	for _, msg := range msgs {
+		counter, err = decode(msg)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return counter, err
+}
+
+// List is part of the interface.
+func (r *Runner) List() ([]*Counter, error) {
+	req := r.newRequest(cmdList)
+	msgs, err := r.execute(req)
+	if err != nil {
+		return nil, err
+	}
+
+	counters := make([]*Counter, 0)
+	for _, msg := range msgs {
+		counter, err := decode(msg)
+		if err != nil {
+			return nil, err
+		}
+		counters = append(counters, counter)
+	}
+	return counters, err
+}
+
+// getAttrTypeAndLen parses length and type of attribute.
+func getAttrTypeAndLen(reader *bytes.Reader) (length uint16, attrType uint16) {
+	_ = binary.Read(reader, binary.NativeEndian, &length)
+	length -= nl.SizeofNfattr
+	_ = binary.Read(reader, binary.NativeEndian, &attrType)
+	return
+}
+
+// decode the netlink nfacct message
+// components of message:
+//   - netfilter header [4 bytes]
+//   - attributes	   [ variable sized, rounded to nearest multiple of 4 bytes]
+//
+// components of attribute:
+//   - length 	[2 bytes]
+//     length includes bytes to define the length itself, bytes to define type
+//     and actual bytes of data without any padding.
+//   - type		[2 bytes]
+//   - data   	[variable sized]
+//   - padding  [optional]
+//     attribute is always rounded up to nearest multiple of 4.
+func decode(msg []byte) (*Counter, error) {
+	counter := &Counter{}
+	reader := bytes.NewReader(msg)
+
+	// skip the first 4 bytes (netfilter header)
+	if _, err := reader.Seek(nl.SizeofNfgenmsg, io.SeekCurrent); err != nil {
+		return nil, err
+	}
+
+	// parse all attributes
+	for reader.Len() > 0 {
+		// get length and type of attribute
+		length, attrType := getAttrTypeAndLen(reader)
+
+		// construct the counter based on attribute
+		switch attrType {
+		case attrName:
+			// read name
+			data := make([]byte, length)
+			_ = binary.Read(reader, binary.NativeEndian, data)
+			counter.Name = string(bytes.TrimRight(data, "\x00"))
+
+			if length%4 != 0 {
+				// skip the padding to align with fixed size length which
+				// is always a multiple of 4.
+				// e.g. if length is 9 then we need to skip the padding of 3 bytes
+				padding := int64((length/4+1)*4 - length)
+				if _, err := reader.Seek(padding, io.SeekCurrent); err != nil {
+					return nil, err
+				}
+			}
+		case attrPackets:
+			// read packets
+			var data uint64
+			if err := binary.Read(reader, binary.BigEndian, &data); err != nil {
+				return nil, err
+			}
+			counter.Packets = data
+		case attrBytes:
+			// read bytes
+			var data uint64
+			if err := binary.Read(reader, binary.BigEndian, &data); err != nil {
+				return nil, err
+			}
+			counter.Bytes = data
+		}
+	}
```

Comment:
If there's an unexpected attribute (future new kernel feature) then you will have skipped over the length and type, but then _not_ skipped the data and so we'll be out of sync now...